src/lib.rs

```diff
@@ -385,10 +386,34 @@ impl<const OSC_RAW_BUF_SIZE: usize> Parser<OSC_RAW_BUF_SIZE> {
                     // byte value, this branch is unreachable.
                     _ => unreachable!("invalid opaque sequence kind"),
                 };
-                performer.opaque_start(kind)
-            },
-            Action::OpaquePut => performer.opaque_put(byte),
-            Action::OpaqueEnd => performer.opaque_end(),
+                self.opaque_sequence_kind = Some(kind);
+                match kind {
+                    OpaqueSequenceKind::Sos => performer.sos_start(),
+                    OpaqueSequenceKind::Pm => performer.pm_start(),
+                    OpaqueSequenceKind::Apc => performer.apc_start(),
+                }
```

Comment:
The kind is assigned earlier in a match statement on the byte. It seems pointless to immediately match here again.
Should just call the functions in the first match.